solidity/test/bridge/WalletCoordinator.test.ts

```diff
@@ -1956,83 +1956,1191 @@ describe("WalletCoordinator", () => {
+
+        await restoreSnapshot()
+      })
+
+      context("when redemption is below the min size", () => {
+        it("should revert", async () => {
+          await expect(
+            walletCoordinator.validateRedemptionProposal({
+              walletPubKeyHash,
+              redeemersOutputScripts: [], // Set size to 0.
+              redemptionTxFee: 0, // Not relevant in this scenario.
+            })
+          ).to.be.revertedWith("Redemption below the min size")
+        })
+      })
+
+      context("when redemption is above the min size", () => {
+        context("when redemption exceeds the max size", () => {
+          it("should revert", async () => {
+            const maxSize = await walletCoordinator.redemptionMaxSize()
+
+            // Pick more redemption requests than allowed.
+            const redeemersOutputScripts = new Array(maxSize + 1).fill(
+              createTestRedemptionRequest(walletPubKeyHash).key
+                .redeemerOutputScript
+            )
+
+            await expect(
+              walletCoordinator.validateRedemptionProposal({
+                walletPubKeyHash,
+                redeemersOutputScripts,
+                redemptionTxFee: 0, // Not relevant in this scenario.
+              })
+            ).to.be.revertedWith("Redemption exceeds the max size")
+          })
+        })
+
+        context("when redemption does not exceed the max size", () => {
+          context("when proposed redemption tx fee is invalid", () => {
+            context("when proposed redemption tx fee is zero", () => {
+              it("should revert", async () => {
+                await expect(
+                  walletCoordinator.validateRedemptionProposal({
+                    walletPubKeyHash,
+                    redeemersOutputScripts: [
+                      createTestRedemptionRequest(walletPubKeyHash).key
+                        .redeemerOutputScript,
+                    ],
+                    redemptionTxFee: 0,
+                  })
+                ).to.be.revertedWith("Proposed transaction fee cannot be zero")
+              })
+            })
+
+            context(
+              "when proposed redemption tx fee is greater than the allowed total fee",
+              () => {
+                it("should revert", async () => {
+                  await expect(
+                    walletCoordinator.validateRedemptionProposal({
+                      walletPubKeyHash,
+                      redeemersOutputScripts: [
+                        createTestRedemptionRequest(walletPubKeyHash).key
+                          .redeemerOutputScript,
+                      ],
+                      // Exceed the max per-request fee by one.
+                      redemptionTxFee: bridgeRedemptionTxMaxTotalFee + 1,
+                    })
+                  ).to.be.revertedWith("Proposed transaction fee is too high")
+                })
+              }
+            )
+          })
```

Comment:
What about the per-redemption fee test?

Reply:
OK, I see it's covered in "when there is a request that incurs an unacceptable tx fee share" test later. Maybe we should add a comment to the end of this block?

Reply:
See d668d5ca